[libpas] Add no-shutting-down scavenger mode

Constellation opened the pull request:
#### e9e63c08f316c931077953049dbe477c9ae099ec
<pre>
[libpas] Add no-shutting-down scavenger mode
<a href="https://bugs.webkit.org/show_bug.cgi?id=242371">https://bugs.webkit.org/show_bug.cgi?id=242371</a>

Reviewed by Mark Lam.

This patch adds pas_scavenger_disable_shut_down, which can disable libpas&apos;s scavenger&apos;s shutting down.
We also add Options::libpasSavengeContinuously to flip this flag.

* Source/bmalloc/libpas/src/libpas/pas_scavenger.c:
(scavenger_thread_main):
(pas_scavenger_suspend):
(pas_scavenger_resume):
(pas_scavenger_disable_shut_down):
* Source/bmalloc/libpas/src/libpas/pas_scavenger.h:
* Source/JavaScriptCore/runtime/InitializeThreading.cpp:
(JSC::initialize):
* Source/JavaScriptCore/runtime/OptionsList.h:

Canonical link: <a href="https://commits.webkit.org/252172@main">https://commits.webkit.org/252172@main</a>
</pre>

MenloDorian commented on Source/bmalloc/libpas/src/libpas/pas_scavenger.c:
```diff
@@ -352,19 +353,21 @@ static void* scavenger_thread_main(void* arg)
+            }
         }
-        
+            
```
Remove these empty spaces.

Constellation replied:
Fixed.

MenloDorian reviewed (approved):
r=me.  Please also add a JSC `Options::libpasSavengeContinuously()`, and in `Options::recomputeDependentOptions()`, set `pas_scavenger_is_shut_down_enabled` if `Options::libpasSavengeContinuously()` is true.

Constellation committed:
[libpas] Add no-shutting-down scavenger mode
https://bugs.webkit.org/show_bug.cgi?id=242371

Reviewed by Mark Lam.

This patch adds pas_scavenger_disable_shut_down, which can disable libpas's scavenger's shutting down.
We also add Options::libpasSavengeContinuously to flip this flag.

* Source/bmalloc/libpas/src/libpas/pas_scavenger.c:
(scavenger_thread_main):
(pas_scavenger_suspend):
(pas_scavenger_resume):
(pas_scavenger_disable_shut_down):
* Source/bmalloc/libpas/src/libpas/pas_scavenger.h:
* Source/JavaScriptCore/runtime/InitializeThreading.cpp:
(JSC::initialize):
* Source/JavaScriptCore/runtime/OptionsList.h:

Canonical link: https://commits.webkit.org/252172@main
